cmd/openshift-install/gather.go

```diff
@@ -106,7 +106,11 @@ func runGatherBootstrapCmd(directory string) (string, error) {
 	bootstrap := gatherBootstrapOpts.bootstrap
 	port := 22
 	masters := gatherBootstrapOpts.masters
-	if bootstrap == "" && len(masters) == 0 {
+	if len(masters) > 0 && bootstrap == "" {
+		return "", fmt.Errorf("when control plane IP addresses are provided, the bootstrap IP address is required")
+	}
```

Comment:
I've introduced this somewhat arbitrary requirement. Alternatively we could try to determine the bootstrap IP if only control plane ips are specified. I put this in to maintain some of the prior behavior, but I think this is a super rare case.

I am wondering a bit about the usability here, though. If we start allowing users to input only the bootstrap ip will they get lazy and not enter the control plane ips, not realizing how important having the control plane IPs is?

I guess the solution is to enter a strong warning message

Reply:
I wonder how many non-machine users supply IPs rather than relying on the gathering of the IPs from the terraform results. I suggest that we allow them to omit control plane IPs and warn if they are omitted. We should always require them to specify the bootstrap IP when specifying the control plane IP, though, like you have.

Reply:
So, yeah, exactly what you wrote.

Reply:
ok so I will relax the current code to allow bootstrap ip without control plane